File: calc_pai.py

def f(x):
    return (1 - x ** 2) ** 0.5

def a(n):
    a_sum = 0
    for k in range(1, n + 1):
        a_sum += f(k / n)
    return a_sum / n

def b(n, a):
    b_sum = 0
    b_sum += (f(0) - f(1)) / n
    b_sum += a
    return b_sum
'''
def b(n, a):
    b_sum = 0
    for k in range(0, n):
        b_sum += f(k / n)
    return b_sum / n
'''
n = 100000000
a = a(n)
b = b(n, a)
print("n = {}, {} < π < {}".format(n, a * 4, b * 4))

File: calc_pai.cpp
#include <iostream>
#include <math.h>

double f(double x) {
   return sqrt(1 - pow(x, 2.0));
}

double a(double n) {
   double a_sum = 0;
   for (double k = 1.0; k < n + 1.0; k++) {
      a_sum += f(k / n);
   }
   return a_sum / n;
}

double b(double n) {
   double b_sum = 0;
   for (double k = 0.0; k < n; k++) {
      b_sum += f(k / n);
   }
   return b_sum / n;
}

int main() {
   double n = 100000000;
   double an = a(n);
   double bn = b(n);
   std::cout << "n = " << n << ", " << an * 4.0 << " < π < " << bn * 4.0 << std::endl;
   return 0;
}
